packages/flutter/test/painting/image_provider_test.dart

```diff
@@ -71,4 +71,19 @@ void main() {
       expect(await caughtError.future, true);
     });
   });
+
+  test('ImageProvide.obtainKey errors will be caught', () async {
```

Comment:
Can you confirm that this test fails without your change?

Reply:
Yup.
```
~/C/f/f/p/flutter ❯❯❯ flutter test ./test/painting/image_provider_test.dart                                                                                                                                                       ✘ 1 
Running "flutter packages get" in flutter...                        0.5s
00:04 +4 -1: ImageProvide.obtainKey errors will be caught [E]                                                                                                                                                                         
  Instance of 'Error'
  test/painting/mocks_for_image_cache.dart 105:5          ObtainKeyErrorImageProvider.obtainKey
  package:flutter/src/painting/image_provider.dart 283:5  ImageProvider.resolve
  test/painting/image_provider_test.dart 81:46            main.<fn>
  ===== asynchronous gap ===========================
  dart:async                                              _AsyncAwaitCompleter.completeError
  test/painting/image_provider_test.dart                  main.<fn>
  
00:04 +4 -1: Some tests failed.     
```